### DISCUSSION TITLE
No space left on device

### DISCUSSION THREAD

**Discussion by asunar** [beginner-corp/begin-community#106] [Support] [answered]
I have an existing arc app. 422mb on disk (399mb is is node_modules ( @architect: 299mb)
15 funcs, (3 funcs  1 to 1.8mb, rest of them under 300kb according to AWS Lambda console - code size)
I can deploy my lambdas using `arc deploy`. 

Getting `no space left on device` when attempting to deploy with begin. Looked at the previously reported space issues, not sure what the issue is. Any help is appreciated.

**Comment by ryanblock** [comment, marked as answer]
Sorry to say, this one is fairly straightforward: your project is larger than the temp disk build container has access to (about 500MB including all functions, function deps, and project deps). We are working on a fix but don’t have an ETA at this time. Can you remove some chonky deps from your project?

**Reply by asunar** [reply]
Thanks for the info. The chunkiest dependency -by far- is architect, cannot remove that. Not using any client frameworks like React. My app folder is 475mb. I read that temp disk capacity is [512mb](https://aws.amazon.com/blogs/compute/choosing-between-aws-lambda-data-storage-options-in-web-apps/). Theoretically, it should work.

I think I will stick to deploying with `arc deploy` instead.

**Reply by ryanblock** [reply]
It is not exactly 512MB but is near that (it can be transient depending on the repo). 

Re. Architect: you can get rid of it as a dependency in your project if you have it installed globally, fwiw! Sadly, most of Architect’s chonkiness is actually aws-sdk, which is provided by default in our build container.

That said, more space and ways to deploy are coming soon!

**Reply by asunar** [reply]
Worked, thanks!
